source/crud/tests/unified/insertOne-dots_and_dollars.yml

```diff
@@ -0,0 +1,241 @@
+description: "insertOne-dots_and_dollars"
+
+schemaVersion: "1.0"
+
+createEntities:
+  - client:
+      id: &client0 client0
+      observeEvents: [ commandStartedEvent ]
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &database0Name crud-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collection0Name coll0
+  - collection:
+      id: &collection1 collection1
+      database: *database0
+      collectionName: &collection1Name coll1
+      collectionOptions:
+        writeConcern: { w: 0 }
+
+initialData: &initialData
+  - collectionName: *collection0Name
+    databaseName: *database0Name
+    documents: []
+
+tests:
+  - description: "Inserting document with top-level dollar-prefixed key on 5.0+ server"
+    runOnRequirements:
+      - minServerVersion: "5.0"
+    operations:
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: &dollarPrefixedKey { _id: 1, $a: 1 }
+        expectResult: &insertResult
+          insertedCount: 1
+          insertedId: { $$unsetOrMatches: 1 }
+    expectEvents: &expectEventsDollarPrefixedKey
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                insert: *collection0Name
+                documents:
+                  - *dollarPrefixedKey
+    outcome:
+      - collectionName: *collection0Name
+        databaseName: *database0Name
+        documents:
+          - *dollarPrefixedKey
+
+  - description: "Inserting document with top-level dollar-prefixed key on pre-5.0 server yields server-side error"
+    runOnRequirements:
+      - maxServerVersion: "4.99"
+    operations:
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: *dollarPrefixedKey
+        expectError:
+          isClientError: false
+    expectEvents: *expectEventsDollarPrefixedKey
+    outcome: *initialData
+
+  - description: "Inserting document with top-level dotted key"
```

Comment:
The original scope talked about creating error-expecting tests for pre-5.0 servers, but I learned that older server versions are actually quite permissive in what they accept. Although [3.6 relaxed dot/dollar restrictions at the storage level](https://docs.mongodb.com/v3.6/reference/limits/#Restrictions-on-Field-Names), inserts on pre-3.6 server versions seem to have few restrictions.

When writing these tests, I ran various server versions (2.6, 3.0, 3.4, 3.6, 4.4, and 5.0) to determine boundaries for each test case and whether we should expect success/error. Note: 2.6 is no longer supported by libmongoc/PHPC, so I happened to use the shell in that case.